### PR TITLE
Use pagination for bodhi's API

### PR DESCRIPTION
**Pull request by carlwgeorge**
Bodhi's API allows you to request multiple states, so there is no need to make separate HTTP calls for each one.  However, we might need to make multiple HTTP calls if there are more than 20 releases, because of pagination.  We must check if there are additional pages of results to make sure we don't miss any releases.  For the corresponding test, the JSON has been manipulated to have 10 results per page in order to test the pagination.

### CODE REVIEW

File: fedora_distro_aliases/__init__.py [hunk 1]
```diff
@@ -22,6 +22,20 @@ def bodhi_active_releases():
...
     releases = response_json["releases"]
+    pages = response_json["pages"]
+
+    # Bodhi's API returns results in pages of 20, so we must check if there are
+    # additional pages to make sure we don't miss any releases.
+    if pages > 1:
+        for page in range(2, pages + 1):
+            response = requests.get(
+                bodhi_url,
+                params={"state": states, "page": page}
```
Comment (praiskup) [inline fedora_distro_aliases/__init__.py]: Can we avoid two get() calls?
```python
page = 1
while:
  resp = get(URL, params)
  if page >= resp.pages:
    break
  page += 1

File: fedora_distro_aliases/__init__.py [hunk 2]
```diff
@@ -14,13 +14,14 @@ def bodhi_active_releases():
...
-    releases = []
     states = ["current", "pending", "frozen"]
-    for state in states:
-        url = "{0}?state={1}".format(bodhi_url, state)
-        response = requests.get(url)
-        response.raise_for_status()
-        releases.extend(response.json()["releases"])
+    response = requests.get(
+        bodhi_url,
+        params={"state": states}
```
Comment (praiskup) [inline fedora_distro_aliases/__init__.py]: Good to know this works!

File: fedora_distro_aliases/__init__.py (hunk 1 above)
Comment (carlwgeorge) [reply fedora_distro_aliases/__init__.py]: Good idea, I'll make that adjustment.

**Comment by FrostyX** [comment]
Thank you very much for the PR @carlwgeorge 

> However, we might need to make multiple HTTP calls if there are more than 20 releases, because of pagination

I found some documentation mentioning "rows_per_page (int) – Limit the results to a certain number of rows per page (min:1 max: 100 default: 20)."
https://bodhi.readthedocs.io/en/stable/python_bindings.html

It seems that the max limit isn't 100 but 1000 because this works:

```python
params={"state": states, "page": page, "rows_per_page": 1000}
```

incrementing to 1001 doesn't.


@carlwgeorge since you already wrote the pagination I think we should keep it just in case but what do you think about setting the `rows_per_page` to something higher than the default 20?

**Comment by carlwgeorge** [comment]
When dealing with pagination, my personal preference is to use the default results per page.  The API has set what it considers a reasonable value as the default.  With proper handling of the paged results you don't have to care about how many results are on each page.  Not specifying a custom `rows_per_page` value also means that future changes to the limit won't affect your code.  Consider the following scenarios:

- We ask for 1000 per page which works now, but then later the API is updated to match the docs limit of 100 and our call starts returning `400 Bad Request` (even though we are handling pagination).
- We ask for 100 per page which works now and matches the docs, but then later the API is updated to set a maximum of 50 and our call starts returning `400 Bad Request` (even though we are handling pagination).

Basically, I consider setting `rows_per_page` to a higher than default value a short-term troubleshooting tool, not a long-term solution to rely on.  If it's a blocker for merging this PR then I can make the change, but I'd prefer to not specify `rows_per_page` at all.

**Comment by FrostyX** [comment]
Good reasoning, I will merge the PR as is. Thank you.
